source/configuration.txt

```diff
@@ -11,53 +11,139 @@ Configuration Options
-``DataFrameWriter``) accept options in the form of a ``Map[String,
-String]``.
+To learn more about :mongo-spark:`ReadConfig
+</blob/master/src/main/scala/com/mongodb/spark/config/ReadConfig.scala>`
+or :mongo-spark:`WriteConfig
+</blob/master/src/main/scala/com/mongodb/spark/config/WriteConfig.scala>` objects, refer to the source for these methods.
 
-You can convert custom ``ReadConfig`` or ``WriteConfig`` settings into
-a ``Map`` via the ``asOptions()`` method.
+.. _options-map:
 
-Via System Property
-~~~~~~~~~~~~~~~~~~~
+Using an Options Map
+~~~~~~~~~~~~~~~~~~~~
+
+In the Spark API, the DataFrameReader and DataFrameWriter methods 
+accept options in the form of a ``Map[String, String]``. Options 
+specified this way override any corresponding settings in ``SparkConf``.
+
+.. tip::
+   
+   You can convert custom ``ReadConfig`` or ``WriteConfig`` settings 
+   into a ``Map`` using the ``asOptions()`` method:
+
+   .. code-block:: scala
+      :copyable: false
+   
+      val df = sparkSession.read.format("mongo").options(customReadConfig.asOptions).load()
+
+.. tabs-drivers::
+
+   tabs:
+     - id: java-sync
+       content: |
+
+         To learn more about specifying options with 
+         `DataFrameReader <https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/DataFrameReader.html#option-java.lang.String-boolean->`__ and 
+         `DataFrameWriter <https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/DataFrameWriter.html#option-java.lang.String-boolean->`__, 
+         refer to the Java Spark documentation for the ``.option()`` 
+         method.
+
+     - id: python
+       content: |
+
+         To learn more about specifying options with 
+         `DataFrameReader <https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.sql.DataFrameReader.option.html#pyspark.sql.DataFrameReader.option>`__ and 
+         `DataFrameWriter <https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.sql.DataFrameWriter.option.html#pyspark.sql.DataFrameWriter.option>`__, 
+         refer to the Java Spark documentation for the ``.option()`` 
+         method.
+
+     - id: scala
+       content: |
+
+         To learn more about specifying options with 
+         `DataFrameReader <https://spark.apache.org/docs/latest/api/scala/org/apache/spark/sql/DataFrameReader.html#option(key:String,value:Double):org.apache.spark.sql.DataFrameReader>`__ and 
+         `DataFrameWriter <https://spark.apache.org/docs/latest/api/scala/org/apache/spark/sql/DataFrameWriter.html#option(key:String,value:Double):org.apache.spark.sql.DataFrameWriter[T]>`__, 
+         refer to the Java Spark documentation for the ``.option()`` 
+         method.
+  
+Short-Form Syntax
+`````````````````
+
+Options maps support short-form syntax. You may omit the prefix when 
+specifying an option key string.
+
+.. example::
+
+   The following syntaxes are equivalent to one another:
+
```

Comment:
Add: `spark.mongodb.write.collection` the `write` namespace provides write specific configuration